bin/git-stamp

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+
+
+init_variables() {
+  COMMAND=${0#*-}
+
+  REPLACE=false
+  unset ID
+  unset MSG
+}
+
+
+usage() {
+  cat << EOF
+usage: git ${COMMAND} [<options>] <id> [<messages>]
+
+Options:
+    -r, --replace    replace all previous stamps with same id
+EOF
+}
+
+
+error() {
+  if [[ -n "$1" ]]; then
+    local msg=$( echo "error: $1" | sed 's/\\n/\\n       /g' )
+    echo -e "${msg}" >&2
+  fi
+  usage
+  exit 1
+}
+
+
+stamp() {
+  local commit_msg=$( git log -1 --pretty=%B )
+  local stamp_msg
+  [[ -n "${MSG}" ]] && stamp_msg="${ID} ${MSG}" || stamp_msg="${ID}"
+  
+  if ${REPLACE}; then
+    # remove previous stamps with same ID from the commit message
+    commit_msg=$(
+      echo "${commit_msg}" \
+        | grep --ignore-case --invert-match "^${ID}\b" \
+        | cat --squeeze-blank
+    )
```

Comment:
Indeed this is "unavoidable" the way it is
That's why I show the new commit message as a result, so that people can check that nothing went wrong.
But it's better to be explicit and I'll put a warning in the documentation